aws/cloudformation/cloud_formation_stack.yml.erb

```diff
@@ -168,6 +168,30 @@ Resources:
       - Type: forward
         TargetGroupArn: !Ref ALBTargetGroup
 
+  CloudfrontOnlyListenerRule:
+  Type: AWS::ElasticLoadBalancingV2::ListenerRule
+  Properties:
+    Actions:
+      - Type: forward
+        TargetGroupArn: !Ref ALBTargetGroup
+    Conditions:
+      - Field: http-header
+        HttpHeaderConfig:
+          HttpHeaderName: X-Cdo-Cloudfront-Allow
+          Values:
+            - !Join ['', ['{{resolve:secretsmanager:', !Ref ListenerRuleSecretHeader, ':SecretString:header-value}}' ]]
+
+  ListenerRuleSecretHeader:
+    Type: 'AWS::SecretsManager::Secret'
+    Properties:
+      Name: ListenerRuleSecretHeader
```

Comment:
Do we need a unique Name for the Secret? Maybe prefix `ListenerRule` with the fully qualified domain name of the CloudFront Distribution?